libqtile/backend/base.py

```diff
@@ -253,6 +256,65 @@ def info(self) -> dict[str, Any]:
         """
         return {}
 
+    @property
+    def kept_above(self) -> bool:
+        return False
+
+    @kept_above.setter
+    def kept_above(self, value: bool) -> None:
+        pass
+
```

Comment:
Is this something we're happy for users to use directly, perhaps from within layouts? I think it's good to expose this information via a read-only property, if logic like those in layouts can benefit from it, but by having a setter we're providing two ways for users to `keep_above()` a window (or at least, it looks like that).

If we don't want users to use this property's setter, it's probably more of an implementation detail of how the X11 backend handles Z stacking, and we should move it there. It's not clear yet if it would also be used by windows in the wayland backend.

Also, how does one undo a `keep_above()` using a command?

Reply:
I don't have any particular concerns with users being able to access it directly.

Undoing `keep_above()` is done by calling it again. The command just runs `self.kept_above = not self.kept_above`.

Reply:
In that case I think we should move this property into the X11 backend's Window implementation, as an implementation detail of how it manages Z stacking, and ideally as a private method (if it's only used by that class). I just want to avoid this becoming part of the 'user API' when we have `keep_above()`.